client/src/components/History/Content/Collection/CollectionDescription.test.js

```diff
@@ -11,23 +11,50 @@ describe("CollectionDescription", () => {
         wrapper = mount(CollectionDescription, {
             propsData: {
                 collectionType: "list",
-                elementCount: 10,
             },
             localVue,
         });
     });
 
-    it("check basics", async () => {
-        const details = wrapper.findAll("span");
-        expect(details.at(0).text()).toBe("a list");
-        expect(details.at(1).text()).toBe("with 10 items");
-        await wrapper.setProps({ elementCount: 1 });
-        expect(details.at(1).text()).toBe("with 1 item");
-        await wrapper.setProps({ collectionType: "paired" });
-        expect(details.at(0).text()).toBe("a dataset pair");
+    it("should display expected heterogeneous descriptions", async () => {
+        const HETEROGENEOUS_DATATYPES = ["txt", "csv", "tabular"];
+        await wrapper.setProps({ elementCount: 1, elementsDatatypes: HETEROGENEOUS_DATATYPES });
+        expect(wrapper.text()).toBe("a list with 1 dataset");
+
+        await wrapper.setProps({ elementCount: 2, collectionType: "paired" });
+        expect(wrapper.text()).toBe("a pair with 2 datasets");
+
+        await wrapper.setProps({ elementCount: 10, collectionType: "list" });
+        expect(wrapper.text()).toBe("a list with 10 datasets");
+
+        await wrapper.setProps({ collectionType: "list:paired" });
+        expect(wrapper.text()).toBe("a list with 10 pairs");
+
+        await wrapper.setProps({ collectionType: "list:list" });
+        expect(wrapper.text()).toBe("a list with 10 lists");
+
+        await wrapper.setProps({ collectionType: "other" });
+        expect(wrapper.text()).toBe("a nested list with 10 datasets");
+    });
+
+    it("should display expected homogeneous descriptions", async () => {
+        const EXPECTED_HOMOGENEOUS_DATATYPE = "tabular";
+        await wrapper.setProps({ elementCount: 1, elementsDatatypes: [EXPECTED_HOMOGENEOUS_DATATYPE] });
+        expect(wrapper.text()).toBe(`a list with 1 ${EXPECTED_HOMOGENEOUS_DATATYPE} dataset`);
+
+        await wrapper.setProps({ elementCount: 2, collectionType: "paired" });
+        expect(wrapper.text()).toBe(`a pair with 2 ${EXPECTED_HOMOGENEOUS_DATATYPE} datasets`);
+
+        await wrapper.setProps({ elementCount: 10, collectionType: "list" });
+        expect(wrapper.text()).toBe(`a list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} datasets`);
+
         await wrapper.setProps({ collectionType: "list:paired" });
-        expect(details.at(0).text()).toBe("a list of pairs");
+        expect(wrapper.text()).toBe(`a list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} pairs`);
+
+        await wrapper.setProps({ collectionType: "list:list" });
+        expect(wrapper.text()).toBe(`a list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} lists`);
+
         await wrapper.setProps({ collectionType: "other" });
-        expect(details.at(0).text()).toBe("a nested list");
+        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} datasets`);
```

Comment:
These aren't datasets, `other` would be a deeply nested collection.
```suggestion
        expect(wrapper.text()).toBe(`a nested list with 10 ${EXPECTED_HOMOGENEOUS_DATATYPE} dataset collections`);
```

Reply:
Thanks for the catch Marius, I've rebased with the changes.